### DISCUSSION TITLE
Depending on single header libraries

### DISCUSSION THREAD

**Discussion by LDAP** [mesonbuild/meson#11817] [Q&A] [answered]
Some libraries are distributed as a single header file:

https://github.com/nothings/stb
https://github.com/tinyobjloader/tinyobjloader
...

What is the current best practice to include those as dependency?

**Comment by eli-schwartz** [comment, marked as answer]
For tinyobjloader it provides a pkg-config and a cmake config file, so:

```meson
dependency('tinyobjloader')
```

For stb this is complicated because upstream doesn't really support anything other than copying the header into your own project sources. No installation routines are provided, and there are multiple separate projects in one git repo that each have separate versioning policies -- which per the upstream developer you may want to have different versions of, so even adding it as a git submodule doesn't work, because you can't have multiple versions of the git repo at the same time.

See e.g. https://github.com/nothings/stb/issues/359

**Reply by LDAP** [reply]
Thanks a lot for looking into this!
